### DISCUSSION TITLE
Managing the complexity of alarms on each path

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10295] [Q&A] [answered]
Hi! I asked the question here https://forum.serverless.com/t/creating-alarms-automatically/15950 but didn't receive a response.

My serverless.yml is pretty unwieldy and it looks like https://s.natalian.org/2021-12-01/serverless.yml As you can see my Alarm definitions is more LOC that my function definitions!!

I need a way of generating {warning,critical} alarms for each path aka *Resource* whether it be a 4xx or 5xx error.  Any suggestions please? For example splitting the alarms in a separate file would already be an improvement in my quality of dev life & I don't know how to do that.

**Comment by pgrzesik** [comment, marked as answer]
Hello @kaihendry 👋 It looks like you're using mostly API Gateway errors alarms in your case so probably the plugin https://github.com/ACloudGuru/serverless-plugin-aws-alerts could be helpful there - I don't see an example in main docs, but there's an issue that shows how to configure such alarms -[https://github.com/ACloudGuru/serverless-plugin-aws-alerts/issues/104#issuecomment-527124751

As for sharing things to different configuration files, you could configure your alarms in separate yaml file e.g. `alarms.yml` and use `file` variable source to "import" them to your main config file: https://www.serverless.com/framework/docs/providers/aws/guide/variables#reference-properties-in-other-files

**Reply by kaihendry** [reply]
Thank you for the fast answer!!

Is there a command to see the final serverless.yml ? I'm still confused about the syntax, is it `${file(./alarms.yml)}`?

**Reply by pgrzesik** [reply]
yes, it can be referenced like that for example

you can use `serverless print` command to see the output of your final configuration, after all variables are resolved

**Reply by kaihendry** [reply]
Got this strange error:

```
 Serverless Error ----------------------------------------

  Cannot parse "serverless.yml": can not read an implicit mapping pair; a colon is missed in "/home/hendry/tmp/sls-http-nodejs-alarm/serverless.yml" (34:26)

   31 |       Properties:
   32 |         TopicName: ${self:service}-NotifyMe
   33 |
   34 |     ${file(./alarms.yml)}
  -------------------------------^
```

**Reply by pgrzesik** [reply]
It has to be set to a specific key, e.g. `Resources: ${file(./alarms.yml)}`
